polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -837,11 +837,16 @@ $loading-panel-height: 53px;
 .TableHeading-align-end {
   text-align: right;
 
-  [class*='TooltipContainer'] {
+  [class*='TooltipContainer'],
+  [class*='SortableTableHeadingWithCustomMarkup'] {
```

Comment:
This fixes an issue where a heading with sort and a tooltip doesn't align right properly.

Reply:
| Before | After |
|--------|--------|
| <img width="844" alt="Screenshot 2023-11-22 at 2 30 54 PM" src="https://github.com/Shopify/polaris/assets/7861746/7f30222a-a995-4261-850b-185f68377bf6"> | <img width="843" alt="Screenshot 2023-11-22 at 2 31 14 PM" src="https://github.com/Shopify/polaris/assets/7861746/316cc4f1-c2b9-4181-a007-4a4061d85031"> |